src/services/transaction_reversal.rs

```diff
@@ -15,17 +20,200 @@ pub struct TransactionReversalPayload<'mpesa> {
+    amount: Option<f64>,
+}
+
+impl<'mpesa, Env: ApiEnvironment> TransactionReversalBuilder<'mpesa, Env> {
+    pub fn new(
+        client: &'mpesa Mpesa<Env>,
+        initiator: &'mpesa str,
+    ) -> TransactionReversalBuilder<'mpesa, Env> {
+        TransactionReversalBuilder {
+            client,
+            initiator,
+            command_id: None,
+            transaction_id: None,
+            receiver_party: None,
+            receiver_identifier_type: None,
+            result_url: None,
+            timeout_url: None,
+            remarks: None,
+            occasion: None,
+            amount: None,
+        }
+    }
+
+    /// Adds `CommandId`. Defaults to `CommandId::TransactionReversal` if no value explicitly passed
+    ///
+    /// # Errors
+    /// If `CommandId` is not valid
+    pub fn command_id(mut self, command_id: CommandId) -> Self {
+        self.command_id = Some(command_id);
+        self
+    }
+
+    /// Add the Mpesa Transaction ID of the transaction which you wish to reverse
+    ///
+    /// This is a required field.
+    pub fn transaction_id(mut self, transaction_id: &'mpesa str) -> Self {
+        self.transaction_id = Some(transaction_id);
+        self
+    }
+
+    /// Organization receiving the transaction
+    ///
+    /// This is required field
+    pub fn receiver_party(mut self, receiver_party: &'mpesa str) -> Self {
+        self.receiver_party = Some(receiver_party);
+        self
+    }
+
+    /// Type of organization receiving the transaction
+    ///
+    /// This is required field
+    pub fn receiver_identifier_type(mut self, receiver_identifier_type: &'mpesa str) -> Self {
+        self.receiver_identifier_type = Some(receiver_identifier_type);
+        self
+    }
 
-pub struct TransactionReversalResponse;
+    // Adds `ResultUrl` This is a required field
+    ///
+    /// # Error
+    /// If `ResultUrl` is invalid or not provided
+    pub fn result_url(mut self, result_url: &'mpesa str) -> Self {
+        self.result_url = Some(result_url);
+        self
+    }
+
+    /// Adds `QueueTimeoutUrl` and `ResultUrl`. This is a required field
+    ///
+    /// # Error
+    /// If either `QueueTimeoutUrl` and `ResultUrl` is invalid or not provided
+    pub fn timeout_url(mut self, timeout_url: &'mpesa str) -> Self {
+        self.timeout_url = Some(timeout_url);
+        self
+    }
+    /// Comments that are sent along with the transaction.
+    ///
+    /// This is required field
+    pub fn remarks(mut self, remarks: &'mpesa str) -> Self {
+        self.remarks = Some(remarks);
+        self
+    }
+    /// Occasion of the transaction
+    /// This is an optional Parameter
+    pub fn occasion(mut self, occasion: &'mpesa str) -> Self {
+        self.occasion = Some(occasion);
+        self
+    }
+
+    /// Adds an `amount` to the request
+    ///
+    /// This is a required field
+    pub fn amount(mut self, amount: f64) -> Self {
+        self.amount = Some(amount);
+        self
+    }
```

Comment:
Could you make `amount` generic enough to accept whole numbers and decimals?

```suggestion
    pub fn amount<Number: Into<f64>>(mut self, amount: Number) -> Self {
        self.amount = Some(amount.into());
        self
    }
```

Reply:
Yep